crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -139,6 +253,16 @@ impl<'db> TypeInferenceBuilder<'db> {
     }
 
     fn infer_function_definition_statement(&mut self, function: &ast::StmtFunctionDef) {
```

Comment:
I think we need some module level or struct level documentation explaining why there are two `infer_function_definition` methods. The difference in naming is so subtle and we better establish a convention or this is going to get very confusing. Ideally, these methods wouldn't be in the same builder, but I think that's difficult to accomplish.

Reply:
Yes, I agree. I'll add such documentation in a follow-up PR.

Reply:
https://github.com/astral-sh/ruff/pull/12356